### PR TITLE
Add the E6.5 keyboard to QMK

### PR DESCRIPTION
**Pull request by masterzen**
## Description

The E6.5 is the new 65% keyboard made by Exclusive.
This changeset adds its PCB to QMK, including all the bottown row variants and iso/ansi/split BS.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/exclusive/e65/e65.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+#include "e65.h"
+
+void matrix_init_kb(void) {
+    setPinOutput(B6);
+    matrix_init_user();
+}
+
+void matrix_scan_kb(void) { matrix_scan_user(); }
```
Comment (fauxpark) [inline keyboards/exclusive/e65/e65.c]:
```suggestion
```

These functions can be removed - they do the same thing the core implementations do.

File: keyboards/exclusive/e65/e65.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) { return process_record_user(keycode, record); }
```
Comment (fauxpark) [inline keyboards/exclusive/e65/e65.c]:
```suggestion
```

File: keyboards/exclusive/e65/e65.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+void led_set_kb(uint8_t usb_led) {
+    if (usb_led & (1 << USB_LED_CAPS_LOCK)) {
```
Comment (fauxpark) [inline keyboards/exclusive/e65/e65.c]:
```suggestion
    if (IS_LED_ON(usb_led, USB_LED_CAPS_LOCK)) {
```

File: keyboards/exclusive/e65/rules.mk [hunk 4]
```diff
@@ -0,0 +1,54 @@
+# MCU name
+MCU = atmega32u4
+
+# Processor frequency.
```
Comment (fauxpark) [inline keyboards/exclusive/e65/rules.mk]: Lines 4 to 37 can be removed - QMK provides these as defaults.

File: keyboards/exclusive/e65/readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# E6.5
+
+![E6.5](https://imgur.com/a/ZsSecSo)
```
Comment (fauxpark) [inline keyboards/exclusive/e65/readme.md]:
```suggestion
![E6.5](https://i.imgur.com/WiDtvPN.jpg)
```

**Comment by masterzen** [comment]
@fauxpark , thanks for the review!
I just pushed a fix.

File: keyboards/exclusive/e65/keymaps/masterzen/keymap.c [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+};
+
+// clang-format on
+
+extern rgblight_config_t rgblight_config;
+
+bool     edit = false;
+uint32_t mode;
+uint16_t hue;
```
Comment (drashna) [inline keyboards/exclusive/e65/keymaps/masterzen/keymap.c]: We actually only use uint8_t for hue. 

Also, you could use the `hsv` struct instead.  
or even `rgb_config_t temp_config;` insead.

File: keyboards/exclusive/e65/keymaps/masterzen/keymap.c [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+uint8_t  sat;
+uint8_t  val;
+
+void matrix_init_user(void) { mode = rgblight_config.mode; }
+
+void persistent_default_layer_set(uint16_t default_layer) {
```
Comment (drashna) [inline keyboards/exclusive/e65/keymaps/masterzen/keymap.c]: Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

File: keyboards/exclusive/e65/keymaps/masterzen/keymap.c [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    eeconfig_update_default_layer(default_layer);
+    default_layer_set(default_layer);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        /* layout switcher */
+        case LAY_LIN:
+            if (record->event.pressed) {
+                persistent_default_layer_set(1UL << BASE);
```
Comment (drashna) [inline keyboards/exclusive/e65/keymaps/masterzen/keymap.c]:
```suggestion
                set_single_persistent_default_layer(BASE);
```

File: keyboards/exclusive/e65/keymaps/masterzen/keymap.c [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+            }
+            return false;
+            break;
+        case LAY_OSX:
+            if (record->event.pressed) {
+                persistent_default_layer_set(1UL << OSX);
```
Comment (drashna) [inline keyboards/exclusive/e65/keymaps/masterzen/keymap.c]:
```suggestion
                set_single_persistent_default_layer(OSX);
```

File: keyboards/exclusive/e65/keymaps/masterzen/keymap.c [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+extern rgblight_config_t rgblight_config;
+
+bool     edit = false;
+uint32_t mode;
+uint16_t hue;
+uint8_t  sat;
+uint8_t  val;
+
+void matrix_init_user(void) { mode = rgblight_config.mode; }
```
Comment (drashna) [inline keyboards/exclusive/e65/keymaps/masterzen/keymap.c]: Calling this in matrix user is actually premature.  This setting isn't populated until after the matrix_init, so the value will always be wrong. 

```suggestion
void keyboard_post_init_user(void) { mode = rgblight_config.mode; }
```

Also, if you use the rgblight_config_t structure, you could just do:

```suggestion
void keyboard_post_init_user(void) { temp_config.raw = rgblight_config.raw; }
```

**Comment by masterzen** [comment]
@drashna thanks for all the comments!

This is some old code that I copy/pasted a long time ago from another keyboard and modified to suits my need. 

I'm going to push a new revision with all your suggestions included.

**Comment by masterzen** [comment]
@drashna, new version that fixes your review comment pushed. 
Thanks!

**Comment by drashna** [comment]
> This is some old code that I copy/pasted a long time ago from another keyboard and modified to suits my need.

Oh man, copypasta gets you in soo much trouble. :D  I know from experience. :D

**Comment by masterzen** [comment]
> > This is some old code that I copy/pasted a long time ago from another keyboard and modified to suits my need.
> 
> Oh man, copypasta gets you in soo much trouble. :D I know from experience. :D

I know too. I'm not actively following QMK internal changes, so I'll always miss new ways of doing old things...

**Comment by drashna** [comment]
Well, everything looks good now! 

Thanks!

**Pull request merged**